### PR TITLE
COMMS-324  Unable to update broadcast repeat pattern from 'Once' to 'More than once'

### PR DESCRIPTION
**Pull request by DanielIsMe**

### CODE REVIEW

File: src/lib/components/Start/OnDate.js [hunk 1]
```diff
@@ -25,7 +26,19 @@ const StartOnDate = ({
...
+    readOnly: true,
+  };
+  if(disabled){
+    inputAttributes = {
+      id: `${id}-datetime`,
+      name: 'start.onDate.date',
+      readOnly: true,    
+      disabled: true,
+    };
+  }
```
Comment (alexjidras) [inline src/lib/components/Start/OnDate.js]: @DanielIsMe try to not repeat code please:
```
let inputAttributes = {
    id: `${id}-datetime`,
    name: 'start.onDate.date',
    readOnly: true,
};
if (disabled) {
    inputAttributes = {...inputAttributes, disabled }
}
  ```
  or
  ```
...
if (disabled) {
    inputAttributes.disabled = disabled;
}
  ```
  or simpler:
  ```
  const inputAttributes = {
    id: `${id}-datetime`,
    name: 'start.onDate.date',
    readOnly: true,
    disabled
};
```

File: src/lib/components/End/OnDate.js [hunk 2]
```diff
@@ -26,6 +26,19 @@ const EndOnDate = ({
...
+      readOnly: true,     
+  };
+  if(disabled){
+    inputAttributes = {
+      id: `${id}-datetime`,
+        name: 'end.onDate.date',
+        readOnly: true,     
+        disabled: true,
+    };
+  }
```
Comment (alexjidras) [inline src/lib/components/End/OnDate.js]: do not repeat code please

File: src/demo/App.js [hunk 3]
```diff
@@ -116,6 +123,46 @@ class App extends Component {
...
+                onChange={(event) => {
+                  const editedEvent = {
+                    ...event,
+                    target: {
+                      ...event.target,
+                      value: !this.state.disabled,
+                      name: event.target.name,
+                    },
+                  };
+
```
Comment (alexjidras) [inline src/demo/App.js]: do not edit event please, just use e.target.checked instead of e.target.value

File: src/demo/App.js [hunk 4]
```diff
@@ -29,8 +30,13 @@ class App extends Component {
     this.setState({ isCopied: true });
   };
 
+  handleChangeDisabled = (event) => {    
+    const hasDisabled = event.target.value;
+    this.setState({ disabled: hasDisabled });
+  };
```
Comment (alexjidras) [inline src/demo/App.js]: for checkboxes and radios we should use event.target.checked, not event.target.value

**Review by alexjidras — CHANGES_REQUESTED** [review]

File: src/lib/components/Start/OnDate.js (hunk 1 above)
Comment (DanielIsMe) [reply src/lib/components/Start/OnDate.js]: @alexjidras yes, thanks.  i think the first one work, i tried the 2nd and 3rd, if the inputAttributes added disabled , the  date component alwasy disabled

**Pull request merged**